### PR TITLE
Issue: #5717: SetValue TIMESTAMP Case

### LINKED ISSUE #5717: ODBC driver: input bind parameters does not support timestamp
### What happens?

1、execute sql like ： update test_table set timestamp_column = strptime('20221215101010','%Y%m%d%H%M%S') where ... 
2、bind parameter type is SQL_TYPE_TIMESTAMP
3、sql execute error

in tools/odbc/parameter_descriptor.cpp function ParameterDescriptor::SetValue
switch sql_desc_type not support SQL_TYPE_TIMESTAMP
I add this code then test was passed

`case SQL_TYPE_TIMESTAMP: {
    auto timestamp_struct = Load<SQL_TIMESTAMP_STRUCT>(dataptr);
    value = Value::TIMESTAMP(timestamp_struct.year, timestamp_struct.month, timestamp_struct.day,
                             timestamp_struct.hour, timestamp_struct.minute, timestamp_struct.second, 0);
    break;
}`

### To Reproduce

1、execute sql like ： update test_table set timestamp_column = strptime('20221215101010','%Y%m%d%H%M%S') where ... 
2、bind parameter type is SQL_TYPE_TIMESTAMP

### OS:

linux

### DuckDB Version:

0.6.1

### DuckDB Client:

odbc

### Full Name:

xumaofeng

### Affiliation:

huawei

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by hawkfish**
Plus test.

fixes #5717

### CODE REVIEW

File: tools/odbc/test/isql-test.py [hunk 1]
```diff
@@ -171,6 +171,8 @@ def test(cmd, out=None, err=None, extra_commands=None, input_file=None):
 test("SELECT TIMESTAMP '-8-01-01 00:00:01.5'::VARCHAR", out='0009-01-01 (BC) 00:00:01.5')
 test("SELECT TIMESTAMP '100000-01-01 00:00:01.5'::VARCHAR", out='100000-01-01 00:00:01.5')
 
+test("UPDATE timestamp SET t = strptime('20221215101010','%Y%m%d%H%M%S') WHERE t='2008-01-01 00:00:01'::TIMESTAMP")
```
Comment (krlmlr) [inline tools/odbc/test/isql-test.py:174]: Do we want a test for the result?

**Review by krlmlr — COMMENTED** [review]
I'm not at all familiar with the APIs involved here. Is there no fractional second component?

File: tools/odbc/test/isql-test.py (hunk 1 above)
Comment (hawkfish) [reply tools/odbc/test/isql-test.py:174]: Dunno - I was kind of winging it here. Mostly I just wanted the test to succeed as I can't seem to verify it on my Mac!

**Comment by hawkfish** [comment]
> I'm not at all familiar with the APIs involved here. Is there no fractional second component?

Sorry I was trying to find someone who as worked on this recently. I'll ask in the chat.

**Comment by hawkfish** [comment]
> I'm not at all familiar with the APIs involved here. Is there no fractional second component?

Good question. There is a `fraction` component that is [specified in nanoseconds](https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/c-data-types?view=sql-server-ver16).  I will update the code.

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM

**Pull request merged**